e2e/testcases/invalid_git_branch_test.go

```diff
@@ -58,6 +60,28 @@ func TestInvalidRootSyncBranchStatus(t *testing.T) {
 
 func TestInvalidRepoSyncBranchStatus(t *testing.T) {
 	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipMonoRepo, ntopts.NamespaceRepo(namespaceRepo, configsync.RepoSyncName))
+	nt.T.Cleanup(func() {
+		nn := nomostest.RepoSyncNN(namespaceRepo, configsync.RepoSyncName)
+
+		// Ensure RepoSync is switched back to main branch locally.
+		nt.NonRootRepos[nn].CheckoutBranch(nomostest.MainBranch)
+		// Ensure RepoSync is reset to main branch in the RootSync.
+		obj := nt.RootRepos[configsync.RootSyncName].Get(nomostest.StructuredNSPath(namespaceRepo, configsync.RepoSyncName))
+		uObj, ok := obj.(*unstructured.Unstructured)
```

Comment:
Pretty sure Repo.Get will try to return the typed object if it's in the scheme. So you should get a RepoSync, it just won't have the GVK set.